Feature/frontend plant db integration

Drewbi opened the pull request:
Review #59 first

Commit:
Changed setUser middleware to checkJwt

Commit:
Updated routes to use controllers

Commit:
Changed getToken to not fail when no token is provided

deanvains committed:
feat(authentication.js): get information on users from auth0

Get information on users from auth0

Commit:
feat(login route): made the login route

Commit:
Changed userschema to remove password

Commit:
Removed console.log from login route

Commit:
Merge branch 'feature/admin-auth' of https://github.com/codersforcauses/FoGUWA into feature/drew-frontend-db-integrate

Commit:
improvement(plant markers): switched from plant json to database

re #48

Commit:
improvement(plantdb.json): removed plantdb local file

Commit:
Swapped over routes to use a controller

Commit:
Merge branch 'fix/vicjohn447-plantIcons' of https://github.com/codersforcauses/FoGUWA into feature/drew-frontend-db-integrate

Commit:
Integrated with icon file

JeremiahPinto commented:
Not accepting this until the other PRs are fixed as this has one of them as the base

JeremiahPinto commented on server/authentication.js:
```diff
@@ -1,15 +1,18 @@
-  const tokenMatcher = /(?<=Bearer ).+/
-  return tokenMatcher.exec(tokenString)[0]
+  const tokenMatch = /(?<=Bearer ).+/.exec(tokenString)
+  if (tokenMatch) return tokenMatch[0]
```
use ternary instead of one line `if`

JeremiahPinto commented on server/controllers/flora.js:
```diff
@@ -0,0 +1,38 @@
+  getFlora: res => {
+    return new Promise((resolve, reject) => {
+      Flora.find((err, res) => {
+        if (err) reject(err)
```
use ternary

JeremiahPinto commented on server/controllers/flora.js:
```diff
@@ -0,0 +1,38 @@
+      flora.icon = floraObject.icon
+      flora.instances = floraObject.instances
+      flora.save((err, res) => {
+        if (err) reject(err)
```
ternary

JeremiahPinto commented on server/controllers/flora.js:
```diff
@@ -0,0 +1,38 @@
+  addFlora: floraObject => {
+    return new Promise((resolve, reject) => {
+      const flora = new Flora()
+      flora.name = floraObject.name
```
use a loop and then assign values using `obj[key] = obj2[key]`. Makes it more scalable

JeremiahPinto replied:
Look into `Object.keys(obj)`

Drewbi replied:
Surely this adds a layer of validation as it discards any fields other than the ones we want.

JeremiahPinto commented on server/controllers/flora.js:
```diff
@@ -0,0 +1,38 @@
+    delete update._id
+    return new Promise((resolve, reject) => {
+      Flora.findByIdAndUpdate(id, update, { new: true }, (err, res) => {
+        if (err) reject(err)
```
ternary

JeremiahPinto commented on server/controllers/users.js:
```diff
@@ -0,0 +1,31 @@
+  addUser: userObject => {
+    return new Promise((resolve, reject) => {
+      const user = new User()
+      user.name.first = userObject.name.first
```
same as above

JeremiahPinto commented on server/controllers/users.js:
```diff
@@ -0,0 +1,31 @@
+    delete update.email
+    return new Promise((resolve, reject) => {
+      User.findByIdAndUpdate(id, update, { new: true }, err => {
+        if (err) reject(err)
```
ternary

JeremiahPinto commented on server/routes/flora.js:
```diff
@@ -17,29 +16,28 @@ router.get('/flora/:id', async (req, res, next) => {
-  delete update._id
-  const flora = await updateModel(Flora, req.params.id, update)
+  const flora = await updateFlora(req.params.id, update)
   if (flora) return res.json(flora)
```
ternary

JeremiahPinto commented on server/routes/flora.js:
```diff
@@ -17,29 +16,28 @@ router.get('/flora/:id', async (req, res, next) => {
-router.delete('/flora/:id', setUser, async (req, res, next) => {
+router.delete('/flora/:id', checkJwt, async (req, res, next) => {
   const flora = await Flora.findByIdAndDelete(req.params.id)
   if (flora) res.json(flora)
```
ternary

JeremiahPinto commented on server/routes/users.js:
```diff
@@ -1,56 +1,50 @@
+router.patch('/users/:id', checkJwt, async (req, res) => {
+  if (mongoose.Types.ObjectId.isValid(req.params.id)) {
+    const user = await updateUser(req.params.id, req.body)
+    if (user) return res.json(user)
```
ternary

JeremiahPinto reviewed (changes requested):
Lots of fixes for you to look into @Drewbi

Drewbi commented:
> Lots of fixes for you to look into @Drewbi

Ternary

Commit:
Ternary

Commit:
Error handling

Commit:
Changed login to userinfo to stop clash with auth0 callback

Commit:
Ternary

Commit:
Ternary

Commit:
Improvements for flora

Commit:
Merge branch 'master' of https://github.com/codersforcauses/FoGUWA into feature/drew-frontend-db-integrate

Commit:
Added req res so that res had functions? or wasn't an await, idk but it works

Commit:
Removed unnecessary res

Commit:
Refactored manual field assignment to user object.keys

Commit:
Merge branch 'master' of https://github.com/codersforcauses/FoGUWA into feature/drew-frontend-db-integrate

JeremiahPinto commented on server/authentication.js:
```diff
@@ -23,4 +25,18 @@ const setUser = expressjwt({
+  try {
+    return await axios.get('https://fog-uwa.au.auth0.com/userinfo', config)
+  } catch (error) {
+    consola.log(error)
```
Should probs return/throw errors to the frontend, or will this be done in the future?

JeremiahPinto reviewed (changes requested):
I'm not approving this until the changes from the previous PR gets fixes as it seems to have that code in as well

Commit:
Fixed error handling

Commit:
Merge branch 'master' of https://github.com/codersforcauses/FoGUWA into feature/drew-frontend-db-integrate

Commit:
Merge branch 'feature/drew-popup' of https://github.com/codersforcauses/FoGUWA into feature/drew-frontend-db-integrate

Commit:
Added plant loading back

JeremiahPinto commented on server/authentication.js:
```diff
@@ -1,15 +1,16 @@
-  const tokenMatcher = /(?<=Bearer ).+/
-  return tokenMatcher.exec(tokenString)[0]
+  const tokenMatch = /(?<=Bearer ).+/.exec(tokenString)
+  return tokenMatch ? tokenMatch[0] : null
```
If `tokenMatch` is an array, it could possibly return an empty array instead of a falsy value which can cause issues. Check for the array length as well just to be safe

JeremiahPinto commented on server/routes/flora.js:
```diff
@@ -1,46 +1,47 @@
+    const flora = await addFlora(req.body)
+    return res.json(flora)
+  } catch (err) {
+    return res.json(err.message)
```
Should send the error code as well, not just the message

JeremiahPinto reviewed (dismissed):
Apart from the fixes to made made from the other PR and the comments below, should modify the seeder to only run in dev

Commit:
fix(token retrival): changed token checker to use length instead of empty array

Commit:
fix(flora post): added error code and message

Drewbi commented:
> should modify the seeder to only run in dev

It already does
https://github.com/codersforcauses/FoGUWA/blob/699bbe9fe8128aa392247f045d98b1abf0027e97/server/index.js#L16-L19

Commit:
Merge branch 'master' of https://github.com/codersforcauses/FoGUWA into feature/drew-frontend-db-integrate

Commit:
Plant images work now lol

Commit:
Updated icon path

Commit:
Fixed icon stroke issue

deanvains committed:
improvement(floraseeds.json): updated Description for Plants

Updated Description for Plants, still waiting on proper description for a few of the plants. Jarrah
plant varieties need a sub description still.

deanvains committed:
improvement(floraseeds.json): fixed the Jarrah sub tree sections

Fixed the Jarrah sub tree sections

deanvains committed:
improvement(flora.js): updated models

updated models

Commit:
Merge branch 'chore/dv-seederUpdate' of https://github.com/codersforcauses/FoGUWA into feature/drew-frontend-db-integrate

Commit:
Merge branch 'master' of https://github.com/codersforcauses/FoGUWA into feature/drew-frontend-db-integrate

Commit:
Removed login route

Drewbi commented on server/models/flora.js:
```diff
@@ -25,6 +25,12 @@ const floraSchema = new mongoose.Schema({
           enum: ['Point'],
           required: true
         },
+        heading: {
```
This needs to be outside location

JeremiahPinto replied:
well, change it lol

JeremiahPinto commented on components/UWAPlantMap.vue:
```diff
@@ -163,7 +171,7 @@ export default {
       markerInstance.addListener('click', () => {
         this.plantInfo = plant
-        this.plantInfo.images = plant.images || defaultInfo.images
+        this.plantInfo.images = plant.images.length > 0 ? plant.images : defaultInfo.images
```
We shouldn't be doing default images

Commit:
Took heading and desc outside location

Commit:
removed default stuff
